### PR TITLE
[MODINVOICE-275] Invoice poNumbers needs to be updated when an invoice line is deleted

### PR DESCRIPTION
**Pull request by aleksandr2110**
## Purpose
 Invoice poNumbers needs to be updated when an invoice line is deleted.

## Approach
Update InvoiceLineHelper#deleteInvoiceLine to add a call to InvoiceLineHelper#updateInvoicePoNumbers.
Update unit tests and cross-module integration tests

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/rest/util/InvoiceHolder.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package org.folio.rest.util;
```
Comment (AndreiMPV) [inline src/main/java/org/folio/rest/util/InvoiceHolder.java:1]: Please put under package org.folio.models

**Review by AndreiMPV — CHANGES_REQUESTED** [review]

**Review by MaksymShtanko — APPROVED** [review]

File: src/main/java/org/folio/rest/impl/InvoiceLineHelper.java [hunk 2]
```diff
@@ -522,6 +537,26 @@ private String getInvoiceLineNumberEndpoint(String id) {
...
+   * @param invoice - the invoice of the modified invoice line
+   * @param invoiceLine - the modified invoice line
+   * @param requestContext - used to start new requests
+   */
+  private CompletableFuture<Void> deleteInvoicePoNumbers(Invoice invoice, InvoiceLine invoiceLine, RequestContext requestContext) {
+
+    if (invoiceLine.getPoLineId() == null)
+      return CompletableFuture.completedFuture(null);
+    return orderService.getPoLine(invoiceLine.getPoLineId(), requestContext)
+      .thenCompose(poLine -> orderService.getOrder(poLine.getPurchaseOrderId(), requestContext))
```
Comment (grabsefx) [inline src/main/java/org/folio/rest/impl/InvoiceLineHelper.java:552]: please use cross-index query for retreiving order by polineId in a single request
`"/orders/composite-orders?query=poLine.id = {{id}}"`

**Review by grabsefx — APPROVED** [review]

File: src/main/java/org/folio/rest/util/InvoiceHolder.java (hunk 1 above)
Comment (aleksandr2110) [reply src/main/java/org/folio/rest/util/InvoiceHolder.java]: Done

**Review by aleksandr2110 — COMMENTED** [review]
fixed

**Review by AndreiMPV — APPROVED** [review]

**Pull request merged**